xdsl/tools/command_line_tool.py

```diff
@@ -117,5 +117,10 @@ def parse_chunk(
                 print(e)
             else:
                 raise
+        except DiagnosticException as e:
+            if self.args.verify_diagnostics:
+                print(e)
+            else:
+                raise
```

Comment:
I couldn't trigger the expected behaviour for the above filechecks without this here. It seems to trigger only upon `apply_passes` invocation, not by just parsing an mlir file.

Worth a separate PR or a workaround I'm missing?

Reply:
CC @math-fehr @alexarice seems like a bug to me, I would expect the tests above to work

Reply:
Seems this catch was missing. In my opinion, we should move the catches for `parsing_diagnostics` and `verify_diagnostics` to `xDSLOptMain.run` so that all the logic is in one place. (Not necessarily in this PR)

Reply:
It's currently a bit of a mess. This function `parse_chunk` is a method of `CommandLineTool` but is referencing an argument `parsing-diagnostics` which is only defined on the subclass `xDSLOptMain`

Reply:
Yeah, the split is a bit all over the place.
I'd suggest opening a separate tiny PR on this, so it unblocks this PR, and we can move on after.

Reply:
@compor Do you want to just merge this then I can try to cook up a quick PR to sort this out on top?

Reply:
You don't need to cook anything, and I'm not in a hurry, I'd rather wait for @math-fehr on this